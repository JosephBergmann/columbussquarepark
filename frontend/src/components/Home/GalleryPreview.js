import ImageCard from "../ImageCard";
import img1 from './gallery_images/image_1.jpeg'
import img2 from './gallery_images/image_2.jpeg'
import img3 from './gallery_images/image_3.jpeg'
import img4 from './gallery_images/image_4.jpeg'
import img5 from './gallery_images/image_5.jpeg'
import img6 from './gallery_images/image_6.jpeg'
import img7 from './gallery_images/image_7.jpeg'


export default function GalleryPreview() {

    const imageMap = [img1, img2, img3, img4, img5, img6, img7].map(image => {
        return (
            <>
            <ImageCard image={image}/>
            </>
        )
    })

    return (
        <>
            {imageMap}
        </>
    )
}
