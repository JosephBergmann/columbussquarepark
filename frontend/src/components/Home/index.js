import parkPhoto from './images/temp_park.jpeg'
import lightbuld from './images/lightbuld.jpeg'
import shovel from './images/shovel_icon.jpeg'
import calendar from './images/calendar_icon.jpeg'
import GalleryPreview from './GalleryPreview'

export default function Home() {

    const subHeaderClass = 'text-center tracking-widest xxs:text-md xs:text-lg sm:text-xl md:text-2xl lg:w-3xl xl:4xl'
    const iconClass = "sm:w-16 md:w-20 lg:w-24 xl:w-32"

    return (
        <>
            <div class="bg-cover bg-center min-h-96" style={{ backgroundImage: `url(${parkPhoto})` }}>
            </div>
            <div className='flex sm:flex-col md:flex-row items-center justify-around gap-20 my-10'>
                <div className='flex flex-col items-center'>
                    <span className={subHeaderClass} >Learn about the park</span>
                    <img className={iconClass} src={lightbuld}/>
                </div>
                <div className='flex flex-col items-center'>
                    <span className={subHeaderClass} >How to help</span>
                    <img className={iconClass} src={shovel}/>
                </div>
                <div className='flex flex-col items-center'>
                    <span className={subHeaderClass} >Upcoming Events</span>
                    <img className={iconClass} src={calendar}/>
                </div>
            </div>
            <div className='flex flex-col items-center'>
                <span className={subHeaderClass}>View our gallery</span>
                <div className='flex justify-around gap-4 mt-4 mb-8'>
                    <GalleryPreview />
                </div>
            </div>
        </>
    )
}
