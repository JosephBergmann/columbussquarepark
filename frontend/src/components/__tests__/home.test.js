import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Home from '../Home'

test('should render home component', () => {
    render(<Home/>)
    const homeElement = screen.getByTestId('home-1');
    expect(homeElement).toBeInTheDocument();
})

test('matches snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
})
