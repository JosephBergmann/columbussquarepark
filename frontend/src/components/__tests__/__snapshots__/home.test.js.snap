// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="home-1"
>
  <div
    class="bg-cover bg-center min-h-96"
    style={
      Object {
        "backgroundImage": "url(temp_park.jpeg)",
      }
    }
  />
  <div
    className="flex sm:flex-col md:flex-row items-center justify-around gap-20 my-10"
  >
    <div
      className="flex flex-col items-center"
    >
      <span
        className="text-center tracking-widest xxs:text-md xs:text-lg sm:text-xl md:text-2xl lg:w-3xl xl:4xl"
      >
        Learn about the park
      </span>
      <img
        className="sm:w-16 md:w-20 lg:w-24 xl:w-32"
        src="lightbuld.jpeg"
      />
    </div>
    <div
      className="flex flex-col items-center"
    >
      <span
        className="text-center tracking-widest xxs:text-md xs:text-lg sm:text-xl md:text-2xl lg:w-3xl xl:4xl"
      >
        How to help
      </span>
      <img
        className="sm:w-16 md:w-20 lg:w-24 xl:w-32"
        src="shovel_icon.jpeg"
      />
    </div>
    <div
      className="flex flex-col items-center"
    >
      <span
        className="text-center tracking-widest xxs:text-md xs:text-lg sm:text-xl md:text-2xl lg:w-3xl xl:4xl"
      >
        Upcoming Events
      </span>
      <img
        className="sm:w-16 md:w-20 lg:w-24 xl:w-32"
        src="calendar_icon.jpeg"
      />
    </div>
  </div>
  <div
    className="flex flex-col items-center"
  >
    <span
      className="text-center tracking-widest xxs:text-md xs:text-lg sm:text-xl md:text-2xl lg:w-3xl xl:4xl"
    >
      View our gallery
    </span>
    <div
      className="flex justify-around gap-4 mt-4 mb-8"
    >
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_1.jpeg"
      />
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_2.jpeg"
      />
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_3.jpeg"
      />
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_4.jpeg"
      />
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_5.jpeg"
      />
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_6.jpeg"
      />
      <img
        className="xxs:max-w-5 xs:min-w-10 sm:min-w-20 md:w-36 lg:w-52 xl:min-w-70 rounded-sm"
        src="image_7.jpeg"
      />
    </div>
  </div>
</div>
`;
